packages/chain-bridge/test/unit/StargateAdapter.refundReentrancy.t.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {StargateAdapter} from "../../src/adapters/StargateAdapter.sol";
+import {IStargateAdapter} from "../../src/interfaces/IStargateAdapter.sol";
+import {BridgeRouterTestHelper} from "../helpers/BridgeRouterTestHelper.sol";
+import {StargateAdapterSetupTest} from "./StargateAdapter.setup.t.sol";
+import {IBridgeAdapter} from "../../src/interfaces/IBridgeAdapter.sol";
+import {BridgeTypes} from "../../src/libraries/BridgeTypes.sol";
+import {Errors} from "@openzeppelin/contracts/utils/Errors.sol";
+
+contract MaliciousRefundReceiver {
+    address public immutable adapter;
+
+    constructor(address _adapter) {
+        adapter = _adapter;
+    }
+
+    receive() external payable {
+        // Attempt to reenter adapter entrypoint.
+        // Refund uses a low gas cap; behavior is gas-sensitive, so adapter standardizes
+        // any refund failure to Errors.FailedCall(). We still exercise a reenter attempt here.
+        // NOTE: Intentionally perform an external call to exceed 2300 gas.
+        (bool s, ) = adapter.call("");
+        s; // silence warning
+        revert("reenter");
```

Comment:
in fact this should NOT revert with `"reenter"` - there is  no check if `s` so it can succeed or fail, if it fails it fill revert with `"reenter"` - if it succeeds it will revert with out of gas.
it should try calling an actual adapter method (or any contract) to fail with out of gas, and if it succeeds then rever with `"reenter"` and that should be considered as a failed test.